File: app/src/main/java/org/movieos/feeder/fragment/LoginFragment.java
package org.movieos.feeder.fragment;

import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import org.movieos.feeder.R;
import org.movieos.feeder.api.Feedbin;
import org.movieos.feeder.databinding.LoginFragmentBinding;
import org.movieos.feeder.utilities.Settings;

import okhttp3.Credentials;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class LoginFragment extends DataBindingFragment<LoginFragmentBinding> {
    @NonNull
    @Override
    protected LoginFragmentBinding createBinding(@NotNull LayoutInflater inflater, @Nullable ViewGroup container) {
        LoginFragmentBinding binding = LoginFragmentBinding.inflate(inflater, container, false);

        binding.button.setOnClickListener(v -> {
            if (getBinding() == null) {
                return;
            }

            String credentials = Credentials.basic(getBinding().email.getText().toString(), getBinding().password.getText().toString());

            getBinding().button.setEnabled(false);
            getBinding().passwordWrapper.setError(null);

            Feedbin.Companion.authenticate(getActivity(), credentials, new Callback<Void>() {
                    @Override
                    public void onResponse(Call<Void> call, Response<Void> response) {
                        if (response.code() == 200) {
                            Settings.INSTANCE.saveCredentials(getActivity(), credentials);
                            getFragmentManager().beginTransaction()
                                .replace(R.id.main_content, new EntriesFragment())
                                .commit();

                        } else {
                            getBinding().passwordWrapper.setError("Authentication failed");
                            getBinding().button.setEnabled(true);
                        }
                    }

                    @Override
                    public void onFailure(Call<Void> call, Throwable t) {
                        getBinding().passwordWrapper.setError("Network error: " + t.getLocalizedMessage());
                        getBinding().button.setEnabled(true);
                    }
                });
        });
        return binding;
    }


}

File: app/src/main/java/org/movieos/feeder/fragment/LoginFragment.kt
package org.movieos.feeder.fragment

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import okhttp3.Credentials
import org.movieos.feeder.R
import org.movieos.feeder.api.Feedbin
import org.movieos.feeder.databinding.LoginFragmentBinding
import org.movieos.feeder.utilities.Settings
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class LoginFragment : DataBindingFragment<LoginFragmentBinding>() {
    override fun createBinding(inflater: LayoutInflater, container: ViewGroup?): LoginFragmentBinding {
        val binding = LoginFragmentBinding.inflate(inflater, container, false)

        binding.button.setOnClickListener { v: View ->
            val credentials = Credentials.basic(binding.email.text.toString(), binding.password.text.toString())

            binding.button.isEnabled = false
            binding.passwordWrapper.error = null

            Feedbin.authenticate(activity, credentials, object : Callback<Void> {
                override fun onResponse(call: Call<Void>, response: Response<Void>) {
                    if (response.code() == 200) {
                        Settings.saveCredentials(activity, credentials)
                        fragmentManager.beginTransaction()
                                .replace(R.id.main_content, EntriesFragment())
                                .commit()

                    } else {
                        binding.passwordWrapper.error = "Authentication failed"
                        binding.button.isEnabled = true
                    }
                }

                override fun onFailure(call: Call<Void>, t: Throwable) {
                    binding.passwordWrapper.error = "Network error: " + t.localizedMessage
                    binding.button.isEnabled = true
                }
            })
        }
        return binding
    }


}
